pkg/contextstore/store.go

```diff
@@ -0,0 +1,290 @@
+package contextstore
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const configFileName = "config.json"
+
+// Store provides a context store for easily remembering endpoints configuration
+type Store interface {
```

Comment:
I think this interface should support `DOCKER_HOST=ssh://` credentials as well. (Implementation can be follow-up PR)

Reply:
Yes, but if I understand it correctly, the ssh support is purely cli-side. So I think we'll solve that in the cli.
And that is one thing I kind of dislike with the ssh support: it seems very difficult for 3rd party clients to leverage it

Reply:
@AkihiroSuda btw, do you think it is possible to modify the ssh implementation to bring it to moby/moby/client ? I did not dig sufficiently into that subject to have a clear vision about it, but that might both simplify things a lot for the context-store integration, and makes it available to 3rd party client tools.

Reply:
Technically yes, although I'm not sure we want to introduce `docker system dial-stdio` dependency on this repo

Reply:
It feels like the subject needs a bit more discussion, and on my side a bit more digging into the design of the ssh support. 
What I plan to do is to open PRs on the CLI side without supporting connection helpers, and after that, we can discuss how to make all this fit together

Reply:
FYI the design of the SSH support has a long story.. https://github.com/docker/go-connections/pull/39#issuecomment-312765226